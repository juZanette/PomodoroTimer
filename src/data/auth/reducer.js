const initialState = {
  signed: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SIGN_IN':
      return {
        signed: true,
      }
    default:
      return state
  }
}
