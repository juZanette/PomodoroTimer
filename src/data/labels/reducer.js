const initialState = {
  dialogOpened: false,
  labelToEdit: null,
  formValue: {
    name: null,
    color: null,
  },
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DIALOG_OPENED':
      return {
        ...state,
        dialogOpened: action.dialogOpened,
      }
    case 'SET_LABEL_TO_EDIT':
      return {
        ...state,
        labelToEdit: action.labelToEdit,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    default:
      return state
  }
}
