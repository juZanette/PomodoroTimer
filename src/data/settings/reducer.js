const defaultState = {
  workDuration: 25,
  shortBreakDuration: 5,
  longBreakDuration: 20,
  rounds: 4,
  showTimerInTitle: true,
  showNotifications: true,
}

const initialState = {
  workDuration: null,
  shortBreakDuration: null,
  longBreakDuration: null,
  rounds: null,
  showTimerInTitle: null,
  showNotifications: null,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_WORK_DURATION':
      return {
        ...state,
        workDuration: action.duration,
      }
    case 'SET_SHORT_BREAK_DURATION':
      return {
        ...state,
        shortBreakDuration: action.duration,
      }
    case 'SET_LONG_BREAK_DURATION':
      return {
        ...state,
        longBreakDuration: action.duration,
      }
    case 'SET_ROUNDS':
      return {
        ...state,
        rounds: action.rounds,
      }
    case 'SET_SHOW_TIMER_IN_TITLE':
      return {
        ...state,
        showTimerInTitle: action.showTimerInTitle,
      }
    case 'SET_SHOW_NOTIFICATIONS':
      return {
        ...state,
        showNotifications: action.showNotifications,
      }
    case 'SET_SETTINGS':
      return {
        ...defaultState,
        ...action.settings,
      }

    default:
      return state
  }
}
