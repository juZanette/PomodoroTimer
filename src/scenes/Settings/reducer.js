const initialState = {
  rounds: 4,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_ROUNDS':
      return {
        rounds: action.roundsNumber,
      }
    default:
      return state
  }
}
