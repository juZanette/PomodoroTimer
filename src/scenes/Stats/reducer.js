const initialState = {
  totalSessions: 0,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'INCREMENT_TOTAL_SESSIONS':
      return {
        totalSessions: ++state.totalSessions,
      }
    default:
      return state
  }
}
