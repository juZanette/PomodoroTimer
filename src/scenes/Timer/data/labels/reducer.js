const initialState = {
  dialogOpened: false,
  formDialogOpened: false,
  deleteAlert: {
    opened: false,
    labelToDelete: null,
  },
  labelToEdit: null,
  formValue: null,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DIALOG_OPENED':
      return {
        ...state,
        dialogOpened: action.dialogOpened,
      }
    case 'SET_FORM_DIALOG':
      return {
        ...state,
        formDialogOpened: action.formDialogOpened,
      }
    case 'SET_DELETE_ALERT':
      const { opened, labelToDelete } = action.deleteAlert

      return {
        ...state,
        deleteAlert: {
          opened,
          labelToDelete,
        },
      }
    case 'SET_LABEL_TO_EDIT':
      return {
        ...state,
        labelToEdit: action.labelToEdit,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    default:
      return state
  }
}
