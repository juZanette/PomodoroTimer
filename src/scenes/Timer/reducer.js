const initialState = {
  timerActive: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_TIMER_ACTIVE':
      return {
        timerActive: action.timerActive,
      }
    default:
      return state
  }
}
