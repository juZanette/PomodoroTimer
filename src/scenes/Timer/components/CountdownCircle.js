import React, { useEffect } from 'react'
import styled from 'styled-components'
import CircularProgress from '@material-ui/core/CircularProgress'
import { useSelector, useDispatch } from 'react-redux'
import { setTimeLeft } from '../actions'
import { STATUSES, TYPES } from '../reducer'

const CircleContainer = styled.div`
  position: relative;
  margin: 20px auto 0;
  height: 200px;
  width: 200px;
`

const Circle = styled(CircularProgress)`
  position: absolute;
  top: 0;
`

const Time = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 3rem;
  white-space: pre;
`

const CountdownCircle = () => {
  const { timeLeft, progress, status, type } = useSelector(
    (state) => state.timer
  )

  const { workDuration, shortBreakDuration, longBreakDuration } = useSelector(
    (state) => state.settings
  )

  const dispatch = useDispatch()

  useEffect(() => {
    if (
      status === STATUSES.onHold &&
      type === TYPES.work &&
      workDuration &&
      (!timeLeft || timeLeft.minutes !== workDuration)
    ) {
      dispatch(setTimeLeft({ minutes: workDuration, seconds: 0 }))
    }
  }, [workDuration])

  useEffect(() => {
    if (
      status === STATUSES.onHold &&
      type === TYPES.shortBreak &&
      timeLeft.minutes !== shortBreakDuration
    ) {
      dispatch(setTimeLeft({ minutes: shortBreakDuration, seconds: 0 }))
    }
  }, [shortBreakDuration])

  useEffect(() => {
    if (
      status === STATUSES.onHold &&
      type === TYPES.longBreak &&
      timeLeft.minutes !== longBreakDuration
    ) {
      dispatch(setTimeLeft({ minutes: longBreakDuration, seconds: 0 }))
    }
  }, [longBreakDuration])

  return (
    <CircleContainer>
      <Circle variant="static" value={100} size={200} color="secondary" />
      <Circle
        variant={timeLeft ? 'static' : 'indeterminate'}
        value={progress}
        size={200}
      />
      {timeLeft && (
        <Time>
          {timeLeft.minutes < 10 ? '0' + timeLeft.minutes : timeLeft.minutes}:
          {timeLeft.seconds < 10 ? '0' + timeLeft.seconds : timeLeft.seconds}
        </Time>
      )}
    </CircleContainer>
  )
}

export { CountdownCircle }
