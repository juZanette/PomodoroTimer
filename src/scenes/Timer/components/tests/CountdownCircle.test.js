import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { createMount } from '@material-ui/core/test-utils'
import { CountdownCircle, Time, TimerType } from '../CountdownCircle'
import { STATUSES, TYPES } from '../../data/timer/reducer'

describe('<CountdownCircle />', () => {
  const mockStore = configureMockStore([thunk])

  let store
  let mount
  let wrapper

  beforeEach(() => {
    mount = createMount()

    wrapper = (
      type = TYPES.work,
      status = STATUSES.onHold,
      timeLeft = null,
      progress = 100
    ) => {
      const storeData = {
        timer: {
          timeLeft,
          progress,
          status,
          type,
        },
        settings: {
          workDuration: 25,
          shortBreakDuration: 5,
          longBreakDuration: 20,
        },
      }

      store = mockStore(storeData)
      store.dispatch = jest.fn()

      return mount(
        <Provider store={store}>
          <CountdownCircle />
        </Provider>
      )
    }
  })

  afterEach(() => {
    mount.cleanUp()
  })

  test('should render <CountdownCircle /> correctly', () => {
    expect(wrapper()).toMatchSnapshot()
  })

  test('should display work duration and FOCUS', () => {
    const wrapperRendered = wrapper(TYPES.work, STATUSES.onHold, {
      minutes: 25,
      seconds: 0,
    })

    expect(wrapperRendered.find(Time).text()).toBe('25:00')
    expect(wrapperRendered.find(TimerType).text().toUpperCase()).toBe('FOCUS')
  })

  test('should display short break duration and BREAK', () => {
    const wrapperRendered = wrapper(TYPES.shortBreak, STATUSES.onHold, {
      minutes: 5,
      seconds: 0,
    })

    expect(wrapperRendered.find(Time).text()).toBe('05:00')
    expect(wrapperRendered.find(TimerType).text().toUpperCase()).toBe('BREAK')
  })

  test('should display long break duration and BREAK', () => {
    const wrapperRendered = wrapper(TYPES.longBreak, STATUSES.onHold, {
      minutes: 20,
      seconds: 0,
    })

    expect(wrapperRendered.find(Time).text()).toBe('20:00')
    expect(wrapperRendered.find(TimerType).text().toUpperCase()).toBe('BREAK')
  })
})
