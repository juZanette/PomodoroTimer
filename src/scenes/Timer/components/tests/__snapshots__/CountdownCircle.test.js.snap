// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountdownCircle /> should render <CountdownCircle /> correctly 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(CircularProgress)))
    color="secondary"
    size={200}
    value={100}
    variant="static"
  />
  <Styled(WithStyles(ForwardRef(CircularProgress)))
    color="primary"
    size={200}
    value={100}
    variant="indeterminate"
  />
  <Styled(Styled(MuiBox))>
    <Styled(WithStyles(ForwardRef(Typography)))
      color="textSecondary"
      variant="subtitle2"
    >
      FOCUS
    </Styled(WithStyles(ForwardRef(Typography)))>
  </Styled(Styled(MuiBox))>
</styled.div>
`;
